### PR TITLE
feat(routing): Enable volume split for dynamic routing

### PR DESCRIPTION
**Pull request by Sarthak1799**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Have added a new API for enabling volume split for dynamic routing service

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

```
curl --location --request POST 'http://localhost:8080/account/sarthak1/business_profile/pro_WlO7KCOeDRB8kWk7MSy1/dynamic_routing/set_volume_split?split=20' \
--header 'api-key: dev_iricUDSeFis6RzTGCcYkOt4wmuBcRfjbVbWXbOUO5996boAgztmtfTKN6HNqerjM'
```
Status in DB - 
image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/core/payments.rs [hunk 1]
```diff
@@ -6026,47 +6026,75 @@ where
...
+                    split: crate::consts::DYNAMIC_ROUTING_MAX_VOLUME
+                        - dynamic_routing_config
+                            .dynamic_routing_volume_split
+                            .unwrap_or_default(),
+                };
+            let volume_split_vec = vec![dynamic_split, static_split];
+            let routing_choice =
+                routing::perform_dynamic_routing_volume_split(volume_split_vec, None)
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("failed to perform volume split on dynamic routing")?;
```
Comment (Chethan-rao) [inline crates/router/src/core/payments.rs:6052]:
```suggestion
                    .attach_printable("failed to perform volume split on the routing type")?;
```

File: crates/router_env/src/logger/types.rs [hunk 2]
```diff
@@ -509,6 +509,8 @@ pub enum Flow {
     PaymentsPostSessionTokens,
     /// Payments start redirection flow
     PaymentStartRedirection,
+    /// Volume split for dynamic routing
+    DynamicRoutingVolumeSplit,
```
Comment (Chethan-rao) [inline crates/router_env/src/logger/types.rs:513]: Should we name this `VolumeSplitOnRoutingType`. This makes it seem like the volume split is on dynamic routing algorithms

File: crates/router/src/routes/app.rs [hunk 3]
```diff
@@ -1776,6 +1776,10 @@ impl Profile {
                             web::resource("/toggle")
                                 .route(web::post().to(routing::toggle_elimination_routing)),
                         ),
+                    )
+                    .service(
+                        web::resource("/set_volume_split")
```
Comment (Chethan-rao) [inline crates/router/src/routes/app.rs:1781]: By just looking at the api-contract, client won't be aware that he is setting volume split for dynamic routing type right? Even the query param is just `split`

**Review by Chethan-rao — CHANGES_REQUESTED** [review]

File: crates/router/src/routes/app.rs (hunk 3 above)
Comment (Sarthak1799) [reply crates/router/src/routes/app.rs:1781]: this is scoped under `/dynamic_routing`

**Review by Chethan-rao — APPROVED** [review]

**Review by prajjwalkumar17 — DISMISSED** [review]

**Review by Chethan-rao — DISMISSED** [review]

**Review by prajjwalkumar17 — APPROVED** [review]

**Review by tsdk02 — APPROVED** [review]

File: crates/router/src/core/payments/routing.rs [hunk 4]
```diff
@@ -489,6 +489,36 @@ pub async fn refresh_routing_cache_v1(
     Ok(arc_cached_algorithm)
 }
 
+pub fn perform_dynamic_routing_volume_split(
```
Comment (SanchithHegde) [inline crates/router/src/core/payments/routing.rs:492]: Can we move this behind dynamic routing feature flag?

**Review by SanchithHegde — APPROVED** [review]

**Review by Chethan-rao — APPROVED** [review]

**Review by tsdk02 — APPROVED** [review]

**Pull request merged**